src/main/java/com/joss/conductor/mobile/TestListener.java

```diff
@@ -0,0 +1,45 @@
+package com.joss.conductor.mobile;
+
+import com.joss.conductor.mobile.util.ScreenShotUtil;
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+
+/**
+ * Created on 3/21/17.
+ */
+public class TestListener implements ITestListener {
+
+    public void onTestStart(ITestResult result) {
+
+    }
+
+    public void onTestSuccess(ITestResult result) {
+
+    }
+
+    public void onTestFailure(ITestResult result) {
+        Locomotive locomotive = (Locomotive) result.getInstance();
+        if (locomotive.configuration.screenshotsOnFail()) {
+                ScreenShotUtil.take(locomotive,
+                        result.getTestClass().getName() + "." + result.getMethod().getMethodName(),
+                        result.getThrowable().getMessage());
```

Comment:
Can we add some kind of logging statement here when the test fails? Or is that already handled somewhere else